modules/auxiliary/gather/windows_secrets_dump.rb

```diff
@@ -1105,4 +1124,12 @@ def run
     simple.client.disconnect! if simple&.client.is_a?(RubySMB::Client)
     disconnect
   end
+
+  private
+
+  def krb_enc_key_to_s(data)
+    # TODO: Do we need to 0xffffff74 rc4 hmac to crypto equivalent 23
```

Comment:
where does `0xffffff74` come from? I know we have this 
```
0xffffff76  => Rex::Proto::Kerberos::Crypto::Rc4Hmac, # Negative 138 two's complement
```

Reply:
https://github.com/rapid7/ruby_smb/blob/95ffce90f3fbd2b2d8d00b643e318fc38cce52bd/lib/ruby_smb/dcerpc/samr.rb#L256-L263

Reply:
Looks like it's not a documented constant, but I was able to get this constant when running against a windows 2008 server

Reply:
Looks like it is documented; 0xffffff74 is -140, and from the windows samr spec link on https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-samr/1355fa6b-d097-4ecc-8d5e-75b3a6533e04 to [<24>](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-samr/fa61e5fc-f8fb-4d5b-9695-c724af0c3829#Appendix_A_24)


```
<24> Section 2.2.10.8: When the current domain functional level is DS_BEHAVIOR_WIN2003 or less,
a Windows Server 2008 and later DC includes a KeyType of -140 in each of
KERB_STORED_CREDENTIAL and KERB_STORED_CREDENTIAL_NEW, which is not needed and can be
ignored; it is a dummy type in the supplemental credentials that is not present when the domain
functional level is raised to DS_BEHAVIOR_WIN2008 or greater. The key data is the NT hash of the
password.
```